src/Serilog/Debugging/SelfLog.cs

```diff
@@ -23,14 +23,63 @@ namespace Serilog.Debugging
     /// </summary>
     public static class SelfLog
     {
+        static Action<string> _output;
+
         /// <summary>
         /// The output mechanism for self-log events.
         /// </summary>
         /// <example>
         /// SelfLog.Out = Console.Error;
         /// </example>
         // ReSharper disable once MemberCanBePrivate.Global, UnusedAutoPropertyAccessor.Global
-        public static TextWriter Out { get; set; }
+        [Obsolete("Use SelfLog.Enable(value) and SelfLog.Disable() instead")]
+        public static TextWriter Out
+        {
+            set
+            {
+                if (value != null)
+                    Enable(value);
+                else
+                    Disable();
+            }
+        }
+
+        /// <summary>
+        /// Set the output mechanism for self-log events.
+        /// </summary>
+        /// <param name="output">A synchronized <see cref="TextWriter"/> to which
+        /// self-log events will be written.</param>
+        // ReSharper disable once MemberCanBePrivate.Global
+        public static void Enable(TextWriter output)
+        {
+            if (output == null) throw new ArgumentNullException(nameof(output));
+
+            Enable(m =>
+            {
+                output.WriteLine(m);
+                output.Flush();
+            });
+        }
+
+        /// <summary>
+        /// Set the output mechanism for self-log events.
+        /// </summary>
+        /// <param name="output">An action to invoke with self-log messages.</param>
```

Comment:
Good catch; I've updated it to use 'messages' everywhere, since when we talk about events in Serilog they're mostly structured, and these are deliberately low-key strings :+1: